pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -787,8 +787,13 @@ func (ctrl *PersistentVolumeController) updateClaimStatus(ctx context.Context, c
 			claimClone.Status.Capacity = nil
 			dirty = true
 		}
+		if utilfeature.DefaultFeatureGate.Enabled(features.VolumeAttributesClass) &&
+			claim.Status.CurrentVolumeAttributesClassName != nil {
+			claimClone.Status.CurrentVolumeAttributesClassName = nil
```

Comment:
the pvc's status could be updated with the ClaimLost phase.

```
	// used for PersistentVolumeClaims that lost their underlying
	// PersistentVolume. The claim was bound to a PersistentVolume and this
	// volume does not exist any longer and all data on it was lost.
	ClaimLost PersistentVolumeClaimPhase = "Lost"
```

In this case, we should reset them because the previous values of those fields can not reflect the truth state.

Reply:
I think the feature-gate check should be removed. when the volume doesn't exist, this value is incorrect. @msau42 what do you think?